arch/s390/dfltcc_common.c

```diff
@@ -0,0 +1,80 @@
+/* dfltcc_deflate.c - IBM Z DEFLATE CONVERSION CALL support. */
+
+#include "dfltcc_common.h"
+
+#include <stdio.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdlib.h>
+
+int env_dfltcc_disabled;
+int env_dfltcc_source_date_epoch;
+unsigned long env_dfltcc_level_mask;
+unsigned long env_dfltcc_block_size;
+unsigned long env_dfltcc_block_threshold;
+unsigned long env_dfltcc_dht_threshold;
+unsigned long env_dfltcc_ribm;
+
+static unsigned long xstrtoul(const char *s, unsigned long _default)
+{
+    char *endptr;
+    unsigned long result;
+
+    if (!(s && *s))
+        return _default;
+    errno = 0;
+    result = strtoul(s, &endptr, 0);
+    return (errno || *endptr) ? _default : result;
+}
+
+__attribute__((constructor)) static void init_globals(void)
+{
+    const char *env;
+
+    env = getenv("DFLTCC");
```

Comment:
zlib has secure_getenv() here and in other places.

Reply:
secure_getenv is behind the `#define _GNU_SOURCE` is it ok to pull it in?

Reply:
I think _GNU_SOURCE is okay: https://stackoverflow.com/questions/5582211/what-does-define-gnu-source-imply. The resulting code may (but not necessarily will) be incompatible with musl though.

Reply:
Another possibility is to open-code it using `getauxval(AT_SECURE)`.

Reply:
We shouldn't use environment variables unless absolutely necessary. Just because other projects are using doesn't mean we should use them. There are several security issues related to using environment variables. If we must use environment variables, we should validate all values but silently ignore invalid input instead of crashing.

Availability of `secure_getenv` should be checked at configure time if it is used.